vendor/k8s.io/utils/inotify/inotify_linux.go

```diff
@@ -120,7 +120,11 @@ func (w *Watcher) RemoveWatch(path string) error {
 	}
 	success, errno := syscall.InotifyRmWatch(w.fd, watch.wd)
 	if success == -1 {
-		return os.NewSyscallError("inotify_rm_watch", errno)
+		// when file descriptor or watch descriptor not found, InotifyRmWatch syscall return EINVAL error
+		// if return error, it may lead this path remain in watches and paths map, and no other event can trigger remove action.
+		if errno != syscall.EINVAL {
+			return os.NewSyscallError("inotify_rm_watch", errno)
+		}
```

Comment:
Are we sure that the inotify object is actually removed whenever the EINVAL error occurs? Otherwise, this may introduce a kernel side memory leak.

Reply:
inotify man pages says watch was removed explicitly or automatically (file was deleted, or file system was unmounted)

Reply:
> (file was deleted, or file system was unmounted)

My question was that
are we sure that the EINVAL error occurs only in that cases?
(maybe we can)

https://man7.org/linux/man-pages/man2/inotify_rm_watch.2.html

```
EINVAL The watch descriptor wd is not valid; or fd is not an inotify file descriptor.
```

Reply:
ref: https://man7.org/linux/man-pages/man2/inotify_rm_watch.2.html